### PR TITLE
Merge data challenge

### PR DESCRIPTION
**Pull request by corwintines**
## Description
- Adds a page for merge data challenge flyer at `/merge-data-challenge`
- Adds a page for merge data challenge form at `/merge-data-challenge/apply`
- Adds a thank you page for merge data challenge at `/merge-data-challenge/thank-you`
- Adds functionality for submitting this form

### CODE REVIEW

File: src/pages/api/merge-data-challenge.ts
Comment (corwintines) [inline src/pages/api/merge-data-challenge.ts:73]: @nhsz would I need to change this to a different record type?

**Review by nhsz — DISMISSED** [review]
LGTM! Good job!

**Review by nhsz — APPROVED** [review]

**Review by samajammin — CHANGES_REQUESTED** [review]
Thanks for this @corwintines!

I think something was lost in communication here, perhaps I didn't explain this well enough. You've essentially created the /apply page but not the main landing page.

A few things I think would greatly improve this page, all modeled off the previous academic grants round we did: 
- landing page: https://deploy-preview-189--ecosystem-support.netlify.app/academic-grants
- apply page: https://deploy-preview-189--ecosystem-support.netlify.app/academic-grants/apply

1) We should create the main landing page, using the artwork Tomo created as the main hero:
Merge-Data-Challenge-hero-large
Tomo mentioned you can ping him for any cropping/dimensions needed.

2) The content of this landing page (using the same layout as the academic grants page) should include the information from the e.org flyer: https://github.com/ethereum/ethereum-org-website/pull/7182 - again, check our the [ESP page for academic grants](https://deploy-preview-189--ecosystem-support.netlify.app/academic-grants) compared to the [ethereum.org page for the academics grants](https://ethereum.org/en/community/grants/academic-grants-round/#:~:text=The%20Ethereum%20Foundation%20is%20sponsoring,apply%20can%20be%20found%20below.).

3) move the current merge data challenge page (https://deploy-preview-224--ecosystem-support.netlify.app/merge-data-challenge) to the /apply page (i.e. /merge-data-challenge/apply)

LMK if that makes sense 😄

**Review by nhsz — CHANGES_REQUESTED** [review]
Looks good overall, good job @corwintines! Asking for a couple of changes.

File: src/pages/merge-data-challenge/apply.tsx
Comment (corwintines) [reply src/pages/merge-data-challenge/apply.tsx]: Yes, thanks for catching. Will remove

File: src/pages/merge-data-challenge/index.tsx
Comment (corwintines) [inline src/pages/merge-data-challenge/index.tsx:591]: Make this "Submit a blog post" instead of "Apply" to unify the CTA

File: src/components/UI/MergeDataChallengeHero.tsx
Comment (corwintines) [inline src/components/UI/MergeDataChallengeHero.tsx:23]: Look at making this less transparent for legibility

File: src/pages/merge-data-challenge/index.tsx
Comment (corwintines) [reply src/pages/merge-data-challenge/index.tsx:287]: Have a TODO to add the href here when there is a mainnet announcement.

File: src/pages/merge-data-challenge/index.tsx
Comment (corwintines) [reply src/pages/merge-data-challenge/index.tsx:155]: Should be fixed with the dom issue mentioned above.

File: src/components/layout/MergeDataChallengeLayout.tsx
Comment (corwintines) [reply src/components/layout/MergeDataChallengeLayout.tsx:1]: I think it would be better to refactor all the hero components to be a single component and keep these to layout components for the different layouts to import images and text specific to that layout. What do you think?

File: src/components/forms/api.ts
Comment (wackerow) [inline src/components/forms/api.ts:169]: @corwintines I think we need to grab the `.value` from the referral source here
```suggestion
          canTheEFReachOut: data.canTheEFReachOut === 'Yes',
          howDidYouHearAboutESP: data.howDidYouHearAboutESP.value
```

Comment (wackerow) [reply src/components/forms/api.ts:169]: This should then be a string, allowing the `.trim()` method

**Review by wackerow — CHANGES_REQUESTED** [review]
@corwintines Suggested a fix to the form submission issue. Also a handful of semantic minor suggestions... overall looking great though 💪

File: src/pages/api/merge-data-challenge.ts
Comment (corwintines) [reply src/pages/api/merge-data-challenge.ts:75]: Should be resolved with below fix.

Comment (corwintines) [reply src/pages/api/merge-data-challenge.ts:75]: Thanks for finding it

File: src/pages/merge-data-challenge/index.tsx
Comment (corwintines) [reply src/pages/merge-data-challenge/index.tsx:277]: Ill ask Rodrigo!

File: src/pages/merge-data-challenge/index.tsx
Comment (corwintines) [reply src/pages/merge-data-challenge/index.tsx:246]: Ill ask Rodrigo

**Comment by samajammin** [comment]
@corwintines does this need review/changes? Or are we just waiting on ESP team before merging?

**Comment by corwintines** [comment]
Only thing I can think of that is missing is the upcoming blog post. After we get that, one final review would be good. This has already been reviewed by @nhsz before, so I don't expect much to be needed.

File: src/pages/merge-data-challenge/thank-you.tsx
Comment (wackerow) [inline src/pages/merge-data-challenge/thank-you.tsx:41]:
```suggestion
                href={`mailto:${MERGE_DATA_CHALLENGE_EMAIL_ADDRESS}`}
```

**Review by wackerow — APPROVED** [review]
Nice job @corwintines!

**Comment by corwintines** [comment]
Only changed the background color from approved review, going to merge

**Pull request merged**
